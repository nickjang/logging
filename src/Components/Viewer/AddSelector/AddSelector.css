.add-selector {
  padding: 0.5rem;
  margin: 2px;
}