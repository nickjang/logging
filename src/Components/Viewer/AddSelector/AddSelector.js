import React, { Component } from 'react';
import PropTypes from 'prop-types';
import SelectorContext from '../../../Context/SelectorContext';
import './AddSelector.css';

class Selector extends Component {
  static contextType = SelectorContext;

  handleClick = (e) => {
    e.preventDefault();
    this.context.addSelector(this.props.projectId, this.props.type);
  }

  render() {
    return (
      <input
        type='button'
        value={this.props.buttonText}
        aria-label={this.props.label}
        onClick={(e) => { this.handleClick(e) }} />
    );
  };
}

Selector.defaultProps = {
  type: '',
  label: '',
  buttonText: '',
  projectId: null,
}

Selector.propTypes = {
  type: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
  buttonText: PropTypes.string.isRequired,
  projectId: PropTypes.number.isRequired
}

export default Selector;