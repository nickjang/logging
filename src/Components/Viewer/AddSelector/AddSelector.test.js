import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import AddSelector from './AddSelector';

describe(`AddSelector component`, () => {
  it('renders AddSelector by default', () => {
    const wrapper = shallow(<AddSelector />);
    expect(toJson(wrapper)).toMatchSnapshot()
  });
})