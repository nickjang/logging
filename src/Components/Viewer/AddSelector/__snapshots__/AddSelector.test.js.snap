// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddSelector component renders AddSelector by default 1`] = `
<input
  aria-label=""
  className="lg-btn lg-btn-light add-selector undefined"
  onClick={[Function]}
  type="button"
  value=""
/>
`;
